docs/applications_general.py

```diff
@@ -0,0 +1,211 @@
+#In this example, we use a Python data flow scripting interface to connect
+#PyEmma, a simulator adapter, and some additional Python code to iteratively
+#run and analyze groups of simulations until convergence.
+#
+#The main script body defines and executes the subgraph.
+#
+#One supporting module (`analyze.py`) wraps PyEmma.
+#
+#supporting modules (`gromacs_wrapper.py` and `lammps_wrapper.py`) 
+#wraps command line MD commands with a SCALE-MS wrapper layer.  The results should be in the same format.
+
+from gromacs_wrapper import collect_configurations, structure_to_pdb, make_input, simulate, modify_input
+
+# configuration_list (an array of filenames) and md_inputs (a file
+# name, or list of file names if multiple inputs are needed to
+# intitialize the simulation.
+#
+# In theory, this could be abstracted further as some sort of input bundle.
+# that contains all information needed to star the simulation.
+# 
+# So if one has N simulations, you would start with a list of N input objects.
+# Each object could be tagged as a package of a supported type (GROMACS, LAMMPS, NAMD),
```

Comment:
In this case, `make_input` is actually, `gromacs_wrapper.make_input`, which indicates to the user that it accepts the GROMACS-specific set of input files, and allows for a module-specific object to be produced that we can either convert for other tools or optimize for GROMACS tool consumers. We could add a layer on top that could use a single command and more uniform or generalized set of key words, plus a key word indicating the simulation platform, which would dispatch to tools for known simulators. But I think this would actually reduce utility, since it would fail at run time instead of while checking syntax, and it is at least as clear to a user to import `make_input` from `gromacs` as it is to specify `gromacs` as an argument or function name.

Reply:
OK, that makes sense: each make_input is a function of a XXXX_wrapper module, so it clearly takes inputs of program XXXXX, and is run by the XXX_wrapper.simulate program of that.